### DISCUSSION TITLE
Create User with Custom properties

### DISCUSSION THREAD

**Discussion by YoumnaJ** [OrchardCMS/OrchardCore#15014] [Q&A] [answered]
Hi, I need to know how to create a user in orchard core CMS  with custom properties using CreateUserAsync function 
thanks,

**Comment by hishamco** [comment]
This is a code snippet I used in one of my projects

```csharp
IUser user = new User
{
    UserName = "hishamco",
    Email = "hishamco_2007@hotmail.com"
    IsEnabled = true,
    RoleNames = new List<string> { "Administrators" },
    Properties = new JObject
    {
        { "Property1", "Value1" },
        { "Property2", "Value2" },
   }
};

await _userService.CreateUserAsync(user, "P@ssw0rd", ModelState.AddModelError);
```

**Reply by YoumnaJ** [reply]
thanks, but my problem is that I need to add new properties other than the existing properties when creating  a new user. Can someone provide me with a solution please?

**Reply by hishamco** [reply]
It was there but I though you need a normal user

**Reply by hishamco** [reply]
I updated the snippet above

**Reply by YoumnaJ** [reply]
thank you a lot for your help

**Reply by lampersky** [reply]
@YoumnaJ, Hishamco already provided working solution, but please have a look here:

Create new **ContentType** with stereotype "CustomUserSettings", this way you will be able to edit those custom properties from Orchard:

![image](https://github.com/OrchardCMS/OrchardCore/assets/6403130/619684f1-44d2-4343-a735-32d5cd718b60)

This will look like this inside "Security > Users":

![image](https://github.com/OrchardCMS/OrchardCore/assets/6403130/751da1bb-9fbd-4e51-9f4b-93119de23be0)

and now, you can modify custom properties this way:
![image](https://github.com/OrchardCMS/OrchardCore/assets/6403130/ec200c4d-7842-4632-83d5-adff1fc8a8f6)

**Reply by hishamco** [reply]
Thanks for the follow-up @lampersky I though he is already added a stereotype :)

**Reply by YoumnaJ** [reply]
Thanks @lampersky but I cannot find new Textfield() in my orchard solution

**Reply by hishamco** [reply, marked as answer]
FYI https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.ContentFields/Fields/TextField.cs

**Reply by YoumnaJ** [reply]
Thank you a lot can you send me please how to create userProfile and userProfilePart

**Reply by hishamco** [reply]
I see the `IdGenerator` is another thing, if your current question is solved please mark the suitable answer, then you can start a new question

**Reply by YoumnaJ** [reply]
I need only to know how to create userProfile and UserProfilePart Classes in my orchard solution

**Reply by hishamco** [reply]
Your first question is "Hi, I need to know how to create a user in orchard core CMS with custom properties using CreateUserAsync function
thanks," :)

Feel free to ask many questions as you want in separate thread

**Reply by MarGraz** [reply]
Hi,

@lampersky I followed the instructions you provided [here](https://github.com/OrchardCMS/OrchardCore/discussions/15014#discussioncomment-8065278) using Orchard Core 1.8.2, but **it doesn't work**. When I edit one of the users, the "UserProfile" tab is not shown.

I set the Content Type as shown [here](https://private-user-images.githubusercontent.com/6403130/295197524-619684f1-44d2-4343-a735-32d5cd718b60.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk1NTk5NzEsIm5iZiI6MTcwOTU1OTY3MSwicGF0aCI6Ii82NDAzMTMwLzI5NTE5NzUyNC02MTk2ODRmMS00NGQyLTQzNDMtYTczNS0zMmQ1Y2Q3MThiNjAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDRUMTM0MTExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzJjYWY3MjdhNTQwMDE3MDI2MzAxY2VlNWUyYzZjOWFiODA4OGNhNzAxMmY0NDBkYzVlNjA5MTk5NDBlNTExYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.qQy6Po-li_h-Qyf9-8oEe_W4_zt8AAbVhpMKmZ2Jac4), and I have **added a `Text` content field called "Legal address" to the "UserProfile"** Content Type.

I need to display the custom properties from the Admin panel. Can you help me?
@hishamco do you have any idea?

Thank you

**Reply by hishamco** [reply]
If you set the `CustomSettings` stereotype you will be able to see your additional properties as a new tab

**Reply by MarGraz** [reply]
@hishamco thank you for your reply.

Doesn't work. I have done this steps:

1. Created a new "Content Types" called `UserProfile`;
2. Added stereotype name `CustomUserSettings`;
3. Added a "field" `Text Field` called `LegalAddress` to the `UserProfile`;
4. Saved;
5. Checked for a second tab `UserProfile` in "Security > Users > Admin user (edit)";
6. No second tab present.

Probably there's something missing 😟 
What's the complete procedure?

**Reply by MarGraz** [reply]
Also [this part](https://docs.orchardcore.net/en/latest/docs/reference/modules/Users/CustomUserSettings/#creating-custom-user-settings:~:text=open%20the%20Setting%20menu%20item) of the Custom User Settings guide, is not clear. Which "Setting menu item" I need to open? 🤔

**Reply by MarGraz** [reply]
@hishamco I found the problem. I never activated the feature `Custom User Settings` from the "Features" list 😅

In the [documentation ](https://docs.orchardcore.net/en/latest/docs/reference/modules/Users/CustomUserSettings/#creating-custom-user-settings) is not mentioned to activate the feature `Custom User Settings`. Fortunately, I saw the [video](https://www.youtube.com/watch?v=_ff79hm5PAc) at the bottom of the page, where is explained to activate it.

I will update the documentation and send a pull request.

Thank you

**Reply by hishamco** [reply]
I thought you did :)
